[TFECO-12319] API DOCS For Registry Tagging and Visibility (#2682)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/cloud-docs/api-docs/private-registry/manage-module-versions.mdx

          "id": "mod-yDKMcmJYx9oFrGao",
          "type": "registry-modules"
        }
      },
      "tag-bindings": {
        "links": {

>           "related": "/api/v2/registry-modules/module-version-id/tag-bindings"

Reviewer: replace `module-version-id` with the id from the response payload, in this case: modver-uKQCVs5vAKDmUMyw"

              "id": "mod-yDKMcmJYx9oFrGao",
              "type": "registry-modules"
          }
      },
      "tag-bindings": {
        "links": {

>           "related": "/api/v2/registry-modules/module-version-id/tag-bindings"

Reviewer: replace `module-version-id` with the id from the response payload

          "id": "mod-u8BFFruSTSz8dN4C",
          "type": "registry-modules"
        }
      },
      "tag-bindings": {
        "links": {

>           "related": "/api/v2/registry-modules/module-version-id/tag-bindings"

Reviewer: replace `module-version-id` with the id from the response payload

```json
{
  "data": {
    "id": "mod-u8BFFruSTSz8dN4C",
    "type": "registry-modules",
    "attributes": {
      "name": "module",
      "namespace": "my-cloud-org",
      "version-tags": [
        {
          "version": "0.1.9",
          "tag": "prod"
        },
        {
          "version": "0.1.9",
          "tag": "dev"
        }
      ],
      // ... //
      "revoked-versions": [
          "1.0.0"
      ],
    // ... //
  },
	  "relationships": {
      // ... //
			"tag-bindings": {
        "links": {

>           "related": "/api/v2/registry-modules/module-version-id/tag-bindings"

Reviewer: replace `module-version-id` with the id from the response payload

        }
      }
    }
	}
}
```

## List all tags for a module version

Use this endpoint to list all tags assigned to a module version.

> `GET /api/v2/registry-module-versions/module_version-id/tag-bindings`

Reviewer: should be `:module-version-id`

| Parameter | Description |
| :---- | :---- |
| `:module_name` | The specific module version id whose tags you want to list. |

| Status | Response | Reason |
| :---- | :---- | :---- |
| [200](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200) | [JSON API document](/terraform/cloud-docs/api-docs#json-api-documents) | The request was successful. |
| [401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) | [JSON API error object](http://jsonapi.org/format/#error-objects) | User unauthorized. |
| [404](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404) | [JSON API error object](http://jsonapi.org/format/#error-objects) | The module version does not exist. |

### Sample request

```shell
curl \
  --request GET \
  --header "Authorization: Bearer $TOKEN" \
  https://app.terraform.io/api/v2/registry-module-versions/modver-fdsgeGGR4ed14kyk5/tag-bindings
```

### Sample response

```json
{
  "data": [
    {
      "id": "tb-KjfdsjI79YFndd7e",

> 			"type": "tag-bindings",

Reviewer: nit: formatting is off here

			"attributes": {
				"key": "environment",
				"value": "staging",
				"created-at": "2026-06-11T02:39:03.363Z"
      }
    },
    {
      "id": "tb-fsdDG4jfPHyuC",
			"type": "tag-bindings",
			"attributes": {
				"key": "environment",
				"value": "testing",
				"created-at": "2026-06-11T02:39:03.369Z"
      }
    }
  ],
  "links": {
    "target": "https://app.terraform.io/api/v2/organizations/my-organization/registry-modules/private/my-namespace/module/terraform/version?name=module&version=0.0.3"
  }
}
```

## Add or update tags for a module version

Use this endpoint to add or update tags for a module version in your organization's registry.
-> **Note:** Removing tags from a module version is made via PATCH requests, where the request body will contain only the tag keys and values that are currently needed.

> `PATCH /api/v2/organizations/:organization_name/registry-modules/private/:namespace/:module_name/:module_provider/:module_version`

Reviewer: replace `private` with `:registry_name` here

| :---- | :---- | :---- |
| [200](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200) | [JSON API document](/terraform/cloud-docs/api-docs#json-api-documents) | Successfully added or updated the tag binding. |
| [400](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) | [JSON API error object](http://jsonapi.org/format/#error-objects) | Request body cannot be parsed. |
| [401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) | [JSON API error object](http://jsonapi.org/format/#error-objects) | User not authorized. |
| [404](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404) | [JSON API error object](http://jsonapi.org/format/#error-objects) | The module version does not exist. |
| [422](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) | [JSON API error object](http://jsonapi.org/format/#error-objects) | Unprocessable content - tag key or value have a wrong format. |

### Request body

This PATCH endpoint requires a JSON object with the following properties as a request payload.

| Key path | Type | Default | Description |
| :---- | :---- | :---- | :---- |
| `data.type` | string |  | Must be `"registry-module-versions"`. |

> | `data.attributes.key` | string |  | The tag key. |

Reviewer: This should be data.relationships.tag_bindings.data.attributes. same applies to all the other keys. Needs to be fixed^

```json
{
  "data": {
    "id": "mod-u8BFFruSTSz8dN4C",
    "type": "registry-modules",
    "attributes": {
      "name": "module",
      "namespace": "my-cloud-org",
      "version-tags": [
        {
          "version": "0.1.9",
          "tag": "prod"
        },
        {
          "version": "0.1.9",
          "tag": "dev"
        }
      ],
      // ... //
      "revoked-versions": [
          "1.0.0"
      ],
    // ... //
  },
	  "relationships": {
      // ... //
			"tag-bindings": {
        "links": {
          "related": "/api/v2/registry-modules/mod-u8BFFruSTSz8dN4C/tag-bindings"
        }
      }
    }
	}
}
```

## List all tags for a module version

> Use this endpoint to list all tags assigned to a module version.

Reviewer 2: ```suggestion
## List all tags

Call the following endpoint to list all tags assigned to a module version.
```

`GET /api/v2/registry-module-versions/:module_version_id/tag-bindings`

| Parameter | Description |
| :---- | :---- |

> | `:module_version_id` | The specific module version id whose tag bindings you want to list. |

Reviewer 2: ```suggestion
| `:module_version_id` | The specific module version ID whose tag bindings you want to list. |
```
How do you get the ID? I couldn't find this information to test and verify.

| [404](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404) | [JSON API error object](http://jsonapi.org/format/#error-objects) | The module version does not exist. |

### Sample request

```shell
curl \
  --request GET \
  --header "Authorization: Bearer $TOKEN" \
  https://app.terraform.io/api/v2/registry-module-versions/modver-fdsgeGGR4ed14kyk5/tag-bindings
```

### Sample response

```json
{
  "data": [
    {
      "id": "tb-KjfdsjI79YFndd7e",
      "type": "tag-bindings",
      "attributes": {
        "key": "environment",
        "value": "staging",
        "created-at": "2026-06-11T02:39:03.363Z"
      }
    },
    {
      "id": "tb-fsdDG4jfPHyuC",
      "type": "tag-bindings",
      "attributes": {
        "key": "environment",
        "value": "testing",
        "created-at": "2026-06-11T02:39:03.369Z"
      }
    }
  ],
  "links": {
    "target": "https://app.terraform.io/api/v2/organizations/my-organization/registry-modules/private/my-namespace/module/terraform/version?name=module&version=0.0.3"
  }
}
```

## Add or update tags for a module version

Use this endpoint to add or update tags for a module version in your organization's registry.

> -> **Note:** Removing tags from a module version is made via PATCH requests, where the request body will contain only the tag keys and values that are currently needed.

Reviewer 2: ```suggestion
## Add or update tags

Send a `PATCH` request to the following endpoint to add, remove, or update tags for a module version in your organization's registry.

The request replaces the existing tags with tag keys and values specified in the request body. To remove specific tags, specify only the tag keys and values you want to keep. A request body with empty `key` and `value` attributes  removes all tags.
```

| Parameter | Description |
| :---- | :---- |
| `:organization_name` | The name of the organization the module belongs to. |
| `:registry_name` | Either `public` or `private`. |
| `:module_name` | The name of the module whose version you want to tag. |
| `:module_provider` | Specifies the provider that this module uses. |
| `:module_version` | The module version you want to add or update a tag for. |

| Status | Response | Reason |
| :---- | :---- | :---- |
| [200](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200) | [JSON API document](/terraform/cloud-docs/api-docs#json-api-documents) | Successfully added or updated the tag binding. |
| [400](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) | [JSON API error object](http://jsonapi.org/format/#error-objects) | Request body cannot be parsed. |
| [401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) | [JSON API error object](http://jsonapi.org/format/#error-objects) | User not authorized. |
| [404](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404) | [JSON API error object](http://jsonapi.org/format/#error-objects) | The module version does not exist. |

> | [422](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) | [JSON API error object](http://jsonapi.org/format/#error-objects) | Unprocessable content - tag key or value have a wrong format. |

Reviewer 2: ```suggestion
| [422][] | [JSON API error object][] | HCP Terraform can't process the content because the tag key or value format is incorrect. |
```

### Request body

> This PATCH endpoint requires a JSON object with the following properties as a request payload.

Reviewer 2: ```suggestion
`PATCH` request must include a JSON object with the following properties as a request payload.
```
